modules/shared/src/main/scala/org/tessellation/ext/kryo.scala

```diff
@@ -33,4 +36,18 @@ object kryo {
       KryoSerializer[F].deserialize[A](bytes).liftTo[F]
   }
 
+  type KryoRegistrationId[C] = Int Refined C
+
+  implicit class MapRegistrationId[C1](mapa: Map[Class[_], KryoRegistrationId[C1]]) {
+
+    def union[C2](mapb: Map[Class[_], KryoRegistrationId[C2]]) =
+      mapa.view.mapValues { x =>
+        val y: KryoRegistrationId[C1 Or C2] = x
+        y
+      }.toMap ++
```

Comment:
I am wondering if we can shorten it:

```scala
mapa.view.mapValues(_.asInstanceOf[KryoRegistrationId[C1 Or C2]])
```

Reply:
With current solution it is a little bit more clear why `union` function is created at all IMO. But I agree, it looks like boilerplate. @tbekas , @marcinwadon , @TheMMaciek , any preferences?

Reply:
Please take a look at my other commend regarding that part